### PR TITLE
refactor: Move event deletion methods

### PR DESCRIPTION
**Pull request by atomrc**

### CODE REVIEW

File: app/script/event/EventService.js [hunk 1]
```diff
@@ -134,6 +133,31 @@ z.event.EventService = class EventService {
...
+  /**
+   * Delete an event from a conversation. Duplicates are delete as well.
+   *
+   * @param {string} conversationId - ID of conversation to remove message from
+   * @param {string} eventId - ID of the actual message
+   * @returns {Promise} Resolves with the number of deleted records
+   */
+  deleteEvent(conversationId, eventId) {
+    return this.storageService.db[this.EVENT_STORE_NAME]
+      .where('id')
```
Comment (atomrc) [inline app/script/event/EventService.js:188]: Minor pref improvements here (selecting first using the event id and then filtering with the right conversation, before it was the other way around).

File: test/unit_tests/event/EventServiceSpec.js [hunk 2]
```diff
@@ -274,6 +274,93 @@ describe('z.event.EventService', () => {
     });
   });
 
+  describe('deleteEvent', () => {
```
Comment (atomrc) [inline test/unit_tests/event/EventServiceSpec.js:338]: These are new tests

File: test/unit_tests/event/EventServiceSpec.js [hunk 3]
```diff
@@ -274,6 +274,132 @@ describe('z.event.EventService', () => {
...
+          expect(newEvents.length).toBe(2);
+        })
+        .then(() => eventService.loadPrecedingEvents('other-conversation-id'))
+        .then(otherConversationEvents => {
+          expect(otherConversationEvents.length).toBe(1);
+        });
+    });
+  });
+
+  describe('deleteEvents', () => {
```
Comment (atomrc) [inline test/unit_tests/event/EventServiceSpec.js:387]: These are new tests

File: app/script/event/EventService.js [hunk 4]
```diff
@@ -134,6 +133,45 @@ z.event.EventService = class EventService {
...
+    return this.storageService.db[this.EVENT_STORE_NAME].delete(primaryKey);
+  }
+
+  /**
+   * Delete all events of a conversation.
+   * @param {string} conversationId - Delete events for this conversation
+   * @param {string} [iso_date] - Date in ISO string format as upper bound which events should be removed
+   * @returns {Promise} Resolves when the events was deleted
+   */
+  deleteEvents(conversationId, iso_date) {
```
Comment (gregor) [inline app/script/event/EventService.js]: Let's camelCase the `isoDate` as well

File: app/script/event/EventService.js [hunk 5]
```diff
@@ -134,6 +133,45 @@ z.event.EventService = class EventService {
...
+      .delete();
+  }
+
+  /**
+   * Delete an event from a conversation with the given primary.
+   *
+   * @param {string} primaryKey - ID of the actual message
+   * @returns {Promise} Resolves with the number of deleted records
+   */
+  deleteEventWithKey(primaryKey) {
```
Comment (gregor) [inline app/script/event/EventService.js]: Just a thought: Is this really deletion "with" key or rather "by" key

File: app/script/conversation/ConversationRepository.js [hunk 6]
```diff
@@ -3523,7 +3523,7 @@ z.conversation.ConversationRepository = class ConversationRepository {
    */
   _delete_message(conversation_et, message_et) {
```
Comment (gregor) [inline app/script/conversation/ConversationRepository.js:3524]: Seems like we have to build a proper EventRepository in the next step to move the event service related functions out of the ConversationRepository. Perfect start to slim that one down as well.

File: app/script/event/EventService.js (hunk 5 above)
Comment (atomrc) [reply app/script/event/EventService.js]: Good catch

File: app/script/conversation/ConversationRepository.js (hunk 6 above)
Comment (atomrc) [reply app/script/conversation/ConversationRepository.js:3524]: I cannot agree more with that statement. That said, we already have an `EventRepository` which is... not really small :D

Comment (gregor) [reply app/script/conversation/ConversationRepository.js:3524]: That one should get a different name. It's more of a `NotificationRepository` - which we also have. So maybe a `InboundBackendTrafficNotificationHandlingRepository`

**Pull request merged**